test/src/misc/poll-nonblocking-socket.c

```diff
@@ -0,0 +1,77 @@
+//! filter.py(TARGET_TRIPLE): wasm32-wasip2
+//! add-flags.py(RUN): --wasi=inherit-network=y
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <netdb.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <poll.h>
+#include "test.h"
+
+#define TEST(c) do { \
+	errno = 0; \
+	if (!(c)) \
+		t_error("%s failed (errno = %d)\n", #c, errno); \
+} while(0)
+
+int BUFSIZE = 256;
+
+// See sockets-server.c -- must be running already as a separate executable
+void test_tcp_client() {
+    // Prepare client socket
+    int socket_fd = socket(AF_INET, SOCK_STREAM, 0);
+    TEST(socket_fd != -1);
+
+    // Prepare server socket
+    // Listen on server_port, but never accept a connection
+    int server_port = 4001;
+    int server_socket_fd = socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK, 0);
+    // Bind server to socket
+    struct sockaddr_in server_address;
+    server_address.sin_family = AF_INET;
+    server_address.sin_addr.s_addr = htonl(INADDR_ANY);
+    server_address.sin_port = htons(server_port);
+    TEST(bind(server_socket_fd, (struct sockaddr*)&server_address, sizeof(server_address)) != -1);
+    // Listen on socket
+    char buffer[BUFSIZE];
+    socklen_t client_len = sizeof(struct sockaddr_in);
+    int client_socket_fd = -1;
+    struct sockaddr_in client_address;
+    int32_t bytes_read = 0, total_bytes_read = 0;
+//    TEST(listen(server_socket_fd, 1) != -1);
```

Comment:
Commented-out `listen` here - should be uncommented
```suggestion
    TEST(listen(server_socket_fd, 1) != -1);
```

Reply:
I left this test in a confusing state -- it was supposed to test something else originally, but now it just tests what happens when a connect request fails. So the code for the server doesn't need to be there at all. I fixed that and force-pushed.